pkgs/by-name/ma/maa-assistant-arknights/package.nix

```diff
@@ -0,0 +1,94 @@
+{ lib
+, config
+, callPackage
+, stdenv
+, fetchFromGitHub
+, asio
+, cmake
+, eigen
+, libcpr
+, onnxruntime
+, opencv
+, isBeta ? false
+, cudaSupport ? config.cudaSupport
+, cudaPackages ? { }
+}:
+
+let
+  fastdeploy = callPackage ./fastdeploy-ppocr.nix { };
+  sources = lib.importJSON ./pin.json;
+in
+stdenv.mkDerivation (finalAttr: {
+  pname = "maa-assistant-arknights" + lib.optionalString isBeta "-beta";
+  version = if isBeta then sources.beta.version else sources.stable.version;
+
+  src = fetchFromGitHub {
+    owner = "MaaAssistantArknights";
+    repo = "MaaAssistantArknights";
+    rev = "v${finalAttr.version}";
+    hash = if isBeta then sources.beta.hash else sources.stable.hash;
+  };
+
+  # https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=maa-assistant-arknights
+  postPatch = ''
+    sed -e 's/RUNTIME\sDESTINATION\s\./ /g' \
+        -e 's/LIBRARY\sDESTINATION\s\./ /g' \
+        -e 's/PUBLIC_HEADER\sDESTINATION\s\./ /g' -i CMakeLists.txt
+    sed -e 's/find_package(asio /# find_package(asio /g' \
+        -e 's/asio::asio/ /g' -i CMakeLists.txt
```

Comment:
I'd prefer `substituteInPlace` with `--replace-fail` for regex-free substitutions.